### ISSUE TITLE
Warning spam: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering

### ISSUE THREAD

**Issue by iXab3r** [retejs/rete#669]
## 🕵️‍♂️ Problem Description
When a user interacts with connections (e.g., lifting up an existing connection), the browser console displays repetitive warning messages. While these warnings don't break any functionality, they can flood the console, especially during batch operations.

## 🌍 Environment
- **Rete version:** `2.0.2` (from NPM)

## 🐾 Steps to Reproduce
1. Visit any React-related example:
   - [Rete Basic React Example](https://retejs.org/examples/basic/react)
   - [CodeSandbox Example](https://codesandbox.io/s/rete-js-v2-yrzxe5?from-embed=&file=/src/App.tsx)
2. Interact with an input or output port by clicking on it. 
   > **Note:** You don't need to reestablish the connection; just lifting it is enough.
3. Observe the console for the following warning:

```plaintext
Warning: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task.
    at ConnectionWrapper (https://yrzxe5.csb.app/node_modules/rete-react-plugin/rete-react-plugin.esm.js:176:24)
    at Root (https://yrzxe5.csb.app/node_modules/rete-react-plugin/rete-react-plugin.esm.js:109:23) 
    in ConnectionWrapper

**Comment by Ni55aN** [comment]
https://github.com/retejs/react-plugin/issues/38#issuecomment-1553225290

**Comment by iXab3r** [comment]
Understood, thanks for the quick reply 😄

TL;DR; Known issue, no working solution yet

**Comment by Ni55aN** [comment]
> Known issue, no working solution yet

not even issue :) React.js doesn't have an option to disable this warning, so in this particular case it's not an issue

**Comment by Ni55aN** [comment]
btw, thanks for raising this topic.

So, if we wouldn't have flushSync, we can see this (if there are a lot of nodes e.i. JS uses CPU close to 100%.):

http://127.0.0.1:5173/?template=perf&rows=100

![perf1](https://github.com/retejs/rete/assets/8259641/814b9fa9-96ce-4f27-9e1b-7f4ab703c8e6)

Just got some time and checked possible solutions

1.  flushSync is implemented in `react-reconciler`, but it is bundled (not peer dependency) into `react-dom` package and isn't exposed ❌
2. the warning recommends using microtask ✔️

```ts
queueMicrotask(() => flushSync(f))
```

Luckly, it doesn't cause a gap between node and connections when `flushSync` is called as a separate microtask

![perf12](https://github.com/retejs/rete/assets/8259641/e6490144-5852-4164-97a8-9c912d27d689)

**Comment by Ni55aN** [comment]
Released in v2.0.4

**Comment by iXab3r** [comment]
Thanks a lot for spending your time on fixing this! :)

**Comment by Necmttn** [comment]
Thank you ❤️
